src/iface/neighbor.rs

```diff
@@ -60,23 +61,59 @@ impl<'a> Cache<'a> {
     /// Neighbor entry lifetime, in milliseconds.
     pub(crate) const ENTRY_LIFETIME: Duration = Duration { millis: 60_000 };
 
+    /// Default number of entries in the cache before GC kicks in
+    pub(crate) const GC_THRESH: usize = 1024;
+
     /// Create a cache. The backing storage is cleared upon creation.
     ///
     /// # Panics
     /// This function panics if `storage.len() == 0`.
     pub fn new<T>(storage: T) -> Cache<'a>
             where T: Into<ManagedMap<'a, IpAddress, Neighbor>> {
+
+        Cache::new_with_limit(storage, Cache::GC_THRESH)
+    }
+
+    pub fn new_with_limit<T>(storage: T, gc_thresh: usize) -> Cache<'a>
+            where T: Into<ManagedMap<'a, IpAddress, Neighbor>> {
         let mut storage = storage.into();
         storage.clear();
 
-        Cache { storage, silent_until: Instant::from_millis(0) }
+        Cache { storage, gc_thresh: gc_thresh, silent_until: Instant::from_millis(0) }
+    }
+
+    #[cfg(any(feature = "std", feature = "alloc"))]
+    pub fn run_owned_gc(&mut self, timestamp: Instant) {
+        let new_map = match self.storage {
+            ManagedMap::Borrowed(_) => {
+                unreachable!()
+            }
+            ManagedMap::Owned(ref mut map) => {
+                map.iter_mut()
```

Comment:
Maybe `.into_iter()` so that we don't need to clone values? If the naive form doesn't pass borrow checker, use `swap`.